### PR TITLE
Expose new iterative algorithm in the CLI + added tests

### LINKED ISSUE #1028: Expose new iterative algorithm in the CLI
To align with our ambition of exposing features in our CLI it is unfortunate that the IES2 algorithm is not exposed in the CLI. The algorithm, together with the options exposed in the GUI, should be made available in the CLI.

### PR DESCRIPTION
**Pull request by frode-aarstad**
Resolves #1028

Pretty much cut and paste from code exposing Ensemble Smoother to the CLI.

Take 2

### CODE REVIEW

**Comment by ertomatic** [comment]
Can one of the admins verify this patch?

**Comment by sondreso** [comment]
Jenkins test this please

File: ert_shared/main.py [hunk 1]
```diff
@@ -237,6 +238,41 @@ def get_ert_parser(parser=None):
...
+    iterative_ensemble_smoother_parser = subparsers.add_parser(
+        ITERATIVE_ENSEMBLE_SMOOTHER_MODE,
+        description=iterative_ensemble_smoother_description,
+        help=iterative_ensemble_smoother_description,
+    )
+    iterative_ensemble_smoother_parser.add_argument(
+        "--target-case",
+        type=valid_name_format,
+        required=True,
+        help="The es_mda creates multiple cases for the different "
```
Comment (sondreso) [inline ert_shared/main.py:255]:
```suggestion
        help="The iterative ensemble smoother creates multiple cases for the different "
```

**Comment by jondequinor** [comment]
```bash
ert iterative_ensemble_smoother poly.ert --target-case default
usage: ert iterative_ensemble_smoother [-h] --target-case TARGET_CASE [--realizations REALIZATIONS]
                                       [--current-case CURRENT_CASE] [--verbose] [--color-always] [--disable-monitoring]
                                       [--enable-new-storage]
                                       config
ert iterative_ensemble_smoother: error: argument --target-case: The argument must be a valid string containing a %d and only characters of these types:  Letters:  A-Z  and  a-z   Numbers:  0-9   Underscore:  _   Dash:  &mdash;   Period:  .   Brackets:  &lt;&gt;
```

This might warrant a change to the error message. I think having entity reference in there is too leaky: the error message should be written for CLI and GUI first, not web based documentation.

**Comment by jondequinor** [comment]
Apart from that bad error message above, I am ready to approve this :)

**Comment by frode-aarstad** [comment]
The error strings are also used in the GUI. In particular a QWidget with setTextFormat(Qt.RichText). This puts constraints on the provided string. 

Perhaps the task of fixing this should be extracted in a new issue

**Comment by sondreso** [comment]
Alright, then I think we can move this to a separate issue! Good investigation :detective:

**Comment by jondequinor** [comment]
I would suggest replacing them with their non-entity reference counter parts and escaping the HTML before it goes into the `QWidget`.

**Comment by jondequinor** [comment]
```python
import html
html.escape("&")  # '&amp;'
```

**Comment by sondreso** [comment]
That is a better solution indeed! So don't listen to me :sweat_smile:

**Comment by frode-aarstad** [comment]
Nice solution. 

Now the only issue left is the function 
`
def strip_error_message_and_raise_exception(validated):
    error = validated.message()
    error = re.sub(r"\<[^>]*\>", " ", error)
    raise ArgumentTypeError(error)
`
from ert_shared/main.py is not happy with strings with "<>". The workaround ofc is to use  "> <" .  

Any thoughts?

**Comment by jondequinor** [comment]
I like that workaround =) Though, why are we even printing these out? Is "angle bracket" ambiguous?

File: ert_gui/ertwidgets/validationsupport.py [hunk 2]
```diff
@@ -32,8 +32,8 @@ def __init__(self):
 
     def presentError(self, widget, error):
         assert isinstance(widget, QWidget)
-
-        self._error_widget.setText(ErrorPopup.error_template % error)
+
```
Comment (sondreso) [inline ert_gui/ertwidgets/validationsupport.py:35]: Extra spaces?

File: ert_shared/storage/rdb_api.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def get_response(self, name, realization_index, ensemble_name):
     def get_response_data(self, name, ensemble_name):
         """Load lightweight "bundle" objects using the ORM."""
 
-        bundle = Bundle("response", Response.id, Response.values, Realization.index)
+        bundle = Bundle("response", Response.id, Response.values_ref, Realization.index)
```
Comment (sondreso) [inline ert_shared/storage/rdb_api.py:94]: This is not supposed to be part of this PR is it?

**Pull request closed without merging**